src/fe-gtk/maingui.c

```diff
@@ -1084,6 +1084,12 @@ mg_tab_close_cb (GtkWidget *dialog, gint arg1, session *sess)
 	}
 }
 
+static struct {
+    server *server;
+    char channel[CHANLEN];
+    char key[64];
+} _last_chan = { 0 };
```

Comment:
I think it makes more sense to use a `GQueue` and store a few previous channels rather than a single one. Also don't start variables with `_`.